ui/src/app/workflows/components/workflow-logs-viewer/workflow-logs-viewer.tsx

```diff
@@ -37,7 +37,11 @@ export const WorkflowLogsViewer = ({workflow, nodeId, container, archived}: Work
             .map(e => (!podName ? e.podName + ': ' : '') + e.content + '\n')
             .publishReplay()
             .refCount();
-        const subscription = source.subscribe(() => setLoaded(true), setError);
+        const subscription = source.subscribe(
+            () => setLoaded(true),
+            setError,
+            () => setLoaded(true)
```

Comment:
This last field is `onCompleted`. When the subscription is completed we don't expect any more data and we therefore don't want to show the "Waiting for data" message